scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/simulation/InMemoryCachedEngineStore.java

```diff
@@ -0,0 +1,123 @@
+package gov.nasa.jpl.aerie.scheduler.simulation;
+
+import gov.nasa.jpl.aerie.merlin.driver.CachedEngineStore;
+import gov.nasa.jpl.aerie.merlin.driver.CheckpointSimulationDriver;
+import gov.nasa.jpl.aerie.merlin.driver.SimulationDriver;
+import gov.nasa.jpl.aerie.merlin.driver.MissionModel;
+import gov.nasa.jpl.aerie.merlin.driver.SimulationDriver;
+import gov.nasa.jpl.aerie.merlin.driver.SimulationEngineConfiguration;
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
+import gov.nasa.jpl.aerie.merlin.protocol.types.SerializedValue;
+import org.apache.commons.collections4.map.ListOrderedMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class InMemoryCachedEngineStore implements AutoCloseable, CachedEngineStore {
+  private record CachedEngineMetadata(
+      SimulationEngineConfiguration configuration,
+      Instant creationDate){}
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(InMemoryCachedEngineStore.class);
+  private final ListOrderedMap<CheckpointSimulationDriver.CachedSimulationEngine, CachedEngineMetadata> cachedEngines;
+  private final int capacity;
+  private Duration savedSimulationTime;
+
+  /**
+   *
+   * @param capacity the maximum number of engines that can be stored in memory
+   */
+  public InMemoryCachedEngineStore(final int capacity) {
+    this.cachedEngines = new ListOrderedMap<>();
+    this.capacity = capacity;
+    this.savedSimulationTime = Duration.ZERO;
+  }
+
+  public Duration getTotalSavedSimulationTime(){
+    return savedSimulationTime;
+  }
+
+  @Override
+  public void close() throws Exception {
```

Comment:
Minor: remove generic `throws Exception`, especially since none of the called methods `throw`
```suggestion
  public void close() {
```